substrate/frame/revive/src/exec.rs

```diff
@@ -1061,23 +1021,33 @@ where
 			let value_transferred = frame.value_transferred;
 			let account_id = &frame.account_id.clone();
 
-			// We need to charge the storage deposit before the initial transfer so that
-			// it can create the account in case the initial transfer is < ed.
+			// We need to make sure that the contract's account exists before calling its
+			// constructor.
 			if entry_point == ExportedFunction::Constructor {
 				// Root origin can't be used to instantiate a contract, so it is safe to assume that
 				// if we reached this point the origin has an associated account.
 				let origin = &self.origin.account_id()?;
 
-				frame.nested_storage.charge_instantiate(
-					origin,
-					&account_id,
-					frame.contract_info.get(&account_id),
-					executable.code_info(),
-					self.skip_transfer,
-				)?;
+				let ed = <Contracts<T>>::min_balance();
+				frame.nested_storage.record_charge(&StorageDeposit::Charge(ed));
+				if self.skip_transfer {
+					T::Currency::set_balance(account_id, ed);
+				} else {
+					T::Currency::transfer(origin, account_id, ed, Preservation::Preserve)?;
+				}
+
+				// A consumer is added at account creation and removed it on termination, otherwise
+				// the runtime could remove the account. As long as a contract exists its
+				// account must exist. With the consumer, a correct runtime cannot remove the
+				// account.
+				<System<T>>::inc_consumers(&frame.account_id)?;
```

Comment:
```suggestion
				<System<T>>::inc_consumers(account_id)?;
```
value is moved or we can just do this?